### PR TITLE
Document how to publish custom React Native version

### PR DESCRIPTION
**Pull request by bartolkaruza**
Add documentation page for #653 to the Android Building from Sources doc.

### CODE REVIEW

File: docs/publishing.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+With this goal of a shortlived fork of React Native in mind, you can publish your own version of React Native. The facebook/react-native repository contains all the dependencies required to be used directly as a git dependency, except for the Android React Native library binary (.aar).
+
+ This binary needs to become available in your project's `node_modules/react-native/android` folder or directly in your gradle dependency of your Android app. You can achieve this in one of two ways: Git dependency branch, Android binary dependency through Maven.
+
+ To build the .aar React Native library, you can follow the steps to [build from source](building-from-source.md) first to install all required tooling. Then to build the actual library, you can run the following in the root of your react-native checkout:
+ ```$bash
```
Comment (gengjiawen) [inline docs/publishing.md]: Can you consider make docker part TL, DR. And add links to docker website.

File: docs/publishing.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+./gradlew :ReactAndroid:installArchives --no-daemon
+ ```
+
+Alternatively, if you don't want to 
+install the required toolchain for building from source, you can use a prebuilt docker image to create a react native binary;
+ ```
+ docker run --rm --name rn-build -v $PWD:/pwd -w /pwd gengjiawen/react-native /bin/sh -c "./gradlew installArchives"
```
Comment (gengjiawen) [inline docs/publishing.md]: Change `gengjiawen/react-native` to `reactnativecommunity/react-native-android`

File: docs/publishing.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+### Publishing to a git fork dependency
+Instead of uploading to Maven/Nexus, you can add the binaries built in the previous steps to git, by changing the .gitignore and committing the binaries to your forked branch. This allows you to make your fork into a functioning git dependency for React Native app projects.
+
+If you have changes that you want to actually merge to React Native, make them on another branch first and open a PR. To start making your dependency branch, make sure you are on a 'release/my-forked-release' branch, then merge any commits that you need from yourself or others into this branch. This release branch should never be merged into any other branch. 
+
+```$bash
+# create .aar, then:
+sed -i '' "s|^/android/||g" .gitignore
+git add android
```
Comment (gengjiawen) [inline docs/publishing.md]: Maybe just `git add android --force`.

File: docs/publishing.md (hunk 1 above)
Comment (gengjiawen) [reply docs/publishing.md]: I want to show developers that make a fork is really simple.

**Comment by bartolkaruza** [comment]
@gengjiawen I've updated the PR for your comments. Can you clarify what you meant with add links to Docker website? I now added a single link to docker.com, but maybe you meant a specific page.

**Comment by gengjiawen** [comment]
I think this will be good https://docs.docker.com/install/

**Comment by bartolkaruza** [comment]
@gengjiawen could you check the docker command is correct? I don't have it installed on this machine (disk space :P), so I wasn't able to run it yet.

I think otherwise this PR is complete

**Comment by gengjiawen** [comment]
LGTM. Great work.

**Comment by bartolkaruza** [comment]
@grabbou you mentioned you guys maintain a lot of forks at Callstack, would you mind having a quick look at this one?

**Comment by grabbou** [comment]
Will look at it tomorrow, thanks for CCing us.

**Comment by grabbou** [comment]
@facebook-github-bot shipit

Our setup doesn't rely on Docker - I am taking @gengjiawen approval as a green light for this PR. Great job! And sorry for the lag, looks like Christmas break took longer than expected

**Pull request merged**